inst/extdata/mvp/product-carbon-transition-risk_review.Rmd

```diff
@@ -0,0 +1,308 @@
+of all products that are at high risk, medium risk or low risk due to
+the products’ relative carbon footprint. The assessment is first
+performed on a product-level and can then be aggregated to the
+company-level.
+
+The Product Carbon Transition Risk measures the relative carbon
+footprint per product. As a default option each product will be compared
+to the carbon footprint of every other product. Alternatively, users can
+also choose to compare those products of the same type, so e.g.,
+comparing only raw materials with each other. Products with a higher
+carbon-footprint will also face a higher risk.
+
+After identifying each carbon footprint for one product, the products
+will be ranked according to their carbon footprint. Products in the
+highest percentile (≥70%) will be classified as high transition risk
+products. Products in the medium percentile (between ≥30% and \<70%)
+will be classified as medium transition risk products. Products in the
+lowest percentile (\<30%) will be classified as low transition risk
+products.
+
+After categorisation, we will aggregate all products from the same
+category and set them in relation to all products that the company
+produces. We derive the Product Carbon Transition Risks.
+
+Please note that carbon footprints, and emissions are used equivalently.
+Carbon footprint refers only on the emissions which occur at the
+production stage of the product and not the emissions from the inputs.
+The unit is CO2e in kg.
+
+The goal of the transition risk MVP is to create a first draft of how
+the transition risk indicator for products would be build in code in
+order to convert it into code production easily in the future. This MVP
+should provide the share of products with “low”, “medium”, or “high”
+relative production emissions per company.
+
+In this markdown file the aim is to prepare data so that they contain
+the following:
+
+- a column with production emissions
+- a column indicating the percentile relative to (i) all products with
+  same unit (ii) products in same sector (iii) products in same segment
+- a column indicating whether the product has “low”, “medium” or “high”
+  relative production emissions
+
+The product carbon transition risk indicator consists of 2 main steps:
+
+    - Step 1: Calculating the relative carbon footprint per product
+    - Step 2: Aggregating on the company-level
+
+### Step 1: Calculating the relative carbon footprint per product
+
+## Data
+
+## Prepare data so that they contain the respective columns:
+
+- replace sample activities for companies with “real activities” from
+  Hanish’s sample
+- include absolute score
+- use ecoinvent’s v3.9 data
+
+## 1. Load packages and ecoinvent data
+
+``` {r}
+library(rlang)
+library(fs)
+library(tidyr)
+library(readr)
+library(dplyr)
+library(graphics)
+library(tiltIndicator)
+
+set.seed(123)
+
+data_ecoinvent <- params$input1 %||% 
+    read_csv(pctr_path("input_data", "ecoinvent_activities.csv"))
```

Comment:
This looks a little weird. It's a small tweak to allow me to develop a function that wraps this .Rmd file, so I can test it and eventually refactor the code in this .Rmd file while ensuring I'm not introducing bugs.

``` r
devtools::load_all()
#> ℹ Loading tiltIndicator
options(readr.show_col_types = FALSE)

input1 <- readr::read_csv(pctr_path("input_data", "ecoinvent_activities.csv"))
input2 <- readr::read_csv(pctr_path("input_data", "MVP_sample_dataset.csv"))

pctr_company_scores(input1, input2)
#> # A tibble: 27 × 3
#>    company_id                                                    risk_…¹ risk_…²
#>    <chr>                                                         <chr>     <dbl>
#>  1 powerstartmanufacturinggmbh_00000005460100827307001           low       0.25 
#>  2 powerstartmanufacturinggmbh_00000005460100827307001           medium    0.5  
#>  3 powerstartmanufacturinggmbh_00000005460100827307001           high      0.25 
#>  4 autohausmanfredmullergmbh_00000004907723001                   low       0.3  
#>  5 autohausmanfredmullergmbh_00000004907723001                   medium    0.5  
#>  6 autohausmanfredmullergmbh_00000004907723001                   high      0.2  
#>  7 ebtor_00000005324899621837001                                 low       0.4  
#>  8 ebtor_00000005324899621837001                                 medium    0.4  
#>  9 ebtor_00000005324899621837001                                 high      0.2  
#> 10 bordelandlandwirtschaftlichegesellschaftmbh_00000004871654001 low       0.182
#> # … with 17 more rows, and abbreviated variable names ¹​risk_groups,
#> #   ²​risk_shares
```

<sup>Created on 2023-03-23 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>